dipy/reconst/multi_voxel.py

```diff
@@ -77,6 +95,7 @@ def new_fit(self, data, mask=None, **kwargs):
                         _parallel_fit_worker,
                         chunks,
                         func_args=[single_voxel_with_self],
+                        func_kwargs=func_kwargs,
                         **kwargs,
```

Comment:
I think the problem come from here.

Each chunk has its own `kwargs`  **OR** Each chunk has its common kwargs.  

- `paramap` should check if there is a list of `kwargs` or just one `kwargs` for all chuncks
- Currently, `paramap` do not handle the case when each chunk have its own `kwargs`. Without that, Sam PR can not work. it was possible before with the multivoxel fit.

Reply:
OK - thanks for this. Let me refactor paramap to accomodate sequences of kwargs as well.

Reply:
Implemented in 934b3818240015bb3cb54ab2db9c75433eaed84d

Reply:
Comment on this: the current implementation supports sequences of func_kwargs, but not sequences of func_args, which need to be identical across calls. Also, the number of items in func_kwargs needs to be identical to the number of items in the data sequence (i.e., one set of func_kwargs per voxel). Does that still work for what you need in your PR, @samcoveney? Adding sequence support for func_args is possible and we could come up with ways to support more complicated combinations (e.g., sets of func_kwargs to map to particular indices in the data), but that will complexify the implementation.

Reply:
> the current implementation supports sequences of func_kwargs, but not sequences of func_args

make sense to me and we should keep it like that. 

> Also, the number of items in func_kwargs needs to be identical to the number of items in the data sequence (i.e., one set of func_kwargs per voxel).

make sense also and I suppose you meant one set of func_kwargs per chunck and not voxel

> Does that still work for what you need in your PR, @samcoveney? 

I think it is good like this but let me try @arokem. Thank you for updating this. I will review it tomorrow or thursday.

> Adding sequence support for func_args is possible

we should avoid that

Reply:
Yes, I would need one set of func_kwargs per chunk, thanks